tests/pruners_tests/test_percentile.py

```diff
@@ -159,9 +161,11 @@ def test_get_best_intermediate_result_over_steps(
     trial_nan = optuna.trial.Trial(study, trial_id_nan)
     trial_nan.report(float("nan"), step=0)
     frozen_trial_nan = study._storage.get_trial(trial_id_nan)
-    assert math.isnan(
-        _percentile._get_best_intermediate_result_over_steps(frozen_trial_nan, direction)
-    )
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=RuntimeWarning)
+        assert math.isnan(
+            _percentile._get_best_intermediate_result_over_steps(frozen_trial_nan, direction)
+        )
```

Comment:
[Note] `with warnings.catch_warnings()` is used to suppress a part of the test function.